windows/src/developer/kmconvert/Keyman.Developer.System.KMConvertParameters.pas

```diff
@@ -0,0 +1,107 @@
+unit Keyman.Developer.System.KMConvertParameters;
+
+interface
+
+uses
+  UKeymanTargets;
+
+type
+  TKMConvertMode = (cmImportWindows, cmTemplate);
+
+  TKMConvertParameters = record
+    KLID: string;
+    Destination: string;
+    KeyboardID: string;
+    Prefix: string;
+    Name: string;
+    Copyright: string;
+    Version: string;
+    BCP47Tags: string;
+    Author: string;
+    Targets: TKeymanTargets;
+    Mode: TKMConvertMode;
+  private
+    function CheckParam(name, value: string): Boolean;
+  public
+    procedure WriteUsage;
+    function CheckParams: Boolean;
+  end;
+
+implementation
+
+uses
+  System.SysUtils;
+
+function TKMConvertParameters.CheckParams: Boolean;
+var
+  ModeString: string;
+  i: Integer;
+begin
+  Destination := '.';
+  Copyright := 'Copyright (C)';
+  Version := '1.0';
+
+  if ParamCount < 2 then
+    Exit(False);
+
+  ModeString := LowerCase(ParamStr(1));
+  if ModeString = 'import-windows' then
+    Mode := cmImportWindows
+  else if ModeString = 'template' then
+    Mode := cmTemplate
+  else
+    Exit(False);
+
+  i := 2;
+  while i < ParamCount do
+  begin
+    if not CheckParam(ParamStr(i), ParamStr(i+1)) then
+    begin
+      writeln('Invalid parameter: '+ParamStr(i));
+      Exit(False);
+    end;
+    Inc(i, 2);
+  end;
+
+  Result := True;
+end;
+
+procedure TKMConvertParameters.WriteUsage;
+begin
+  writeln('kmconvert import-windows -klid <source-klid> [additional-options]');
+  writeln('  Imports a Windows keyboard into a new Keyman keyboard project');
+  writeln;
+  writeln('kmconvert template -id <keyboard_id> [additional-options]');
+  writeln('  Creates a blank keyboard project in the repository template format');
+  writeln;
+  writeln('Parameters:');
+  writeln('  -klid <source-klid>  The KLID of the keyboard to import, per LoadKeyboardLayout');
+  writeln('  -o <destination>     The target folder to write the keyboard project into, defaults to "."');
+  writeln('  -id <keyboard_id>    The id of the keyboard to create');
+  writeln('                       (in `import-windows` mode, can be a format string)');
+  writeln('  -name <data>         Name of the keyboard, e.g. "My First Keyboard", "%s Basic" ');
+  writeln('                       (format strings are only valid in `import-windows` mode)');
+  writeln('  -copyright <data>    Copyright string for the keyboard, defaults to "Copyright (C)"');
+  writeln('  -version <data>      Version number of the keyboard, defaults to "1.0"');
+  writeln('  -languages <data>    space-separated list of BCP 47 tags, e.g. "en-US tpi-PG"');
+  writeln('  -author <data>       Name of author of the keyboard, no default');
+end;
+
+{ TKMConvertParameters }
+
+function TKMConvertParameters.CheckParam(name, value: string): Boolean;
+begin
+  if name = '-klid' then KLID := value
+  else if name = '-o' then Destination := value
+  else if name = '-id' then KeyboardID := value
+  else if name = '-prefix' then Prefix := value
```

Comment:
-prefix isn't mentioned in the WriteUsage

Reply:
Good catch